### ISSUE TITLE
Add SHAP parity test — ShapExplainer is the only backend without one

### ISSUE THREAD

**Issue by pizzajojo** [CAIIVS/raitap#248]
## Summary

`ShapExplainer` is the only attribution/attack backend **without** a parity test. The repo has:

- `src/raitap/transparency/explainers/tests/test_captum_parity.py`
- `src/raitap/robustness/assessors/tests/test_torchattacks_parity.py`
- `src/raitap/robustness/assessors/tests/test_foolbox_parity.py`

…but no `test_shap_parity.py`. The existing SHAP tests (`test_shap_explainer.py`, `test_shap_explainer_errors.py`) cover behaviour/errors, not equality against a direct `shap` call.

## Why this matters

Parity tests assert the wrapper returns the *same numbers* as the underlying library called directly — they catch dropped kwargs, wrong target wiring, or stray mutation in the data path. SHAP is the backend that needs it **most**, because it has the most wrapper logic between the library call and the persisted artefact:

- multi-class class-stacking (`torch.stack(..., dim=-1)`, `shap_explainer.py:222-230`)
- per-sample target selection (`_select_target_attributions`, `shap_explainer.py:50-72`)

Captum is a near pass-through and is covered; SHAP does real post-processing and is uncovered.

## Proposed test

A `test_shap_parity.py` mirroring `test_captum_parity.py`, for `GradientExplainer`:

```python
# raitap path
out = ShapExplainer("GradientExplainer").compute_attributions(
    model, x, background_data=bg, target=target)

# bare path: same call + replicate the documented post-processing explicitly
sv = shap.GradientExplainer(model, bg).shap_values(x)
ref = select_target(stack_classes(sv), target)   # stack(dim=-1) + per-sample index

assert torch.allclose(out, ref, rtol=1e-5, atol=1e-6)
```

`GradientExplainer` is stochastic (path sampling), so re-seed (`np.random` + `torch`) immediately before **each** path, the way `test_torchattacks_parity.py` pins reproducibility (`random_start=False`). Replicating the stack+select in the bare path (rather than importing the helper) also documents that step as a pure reshape/index.

## Evidence it should pass

The SHAP wrapper was verified bitwise-identical to a direct `shap.GradientExplainer` call on real models, under a fixed seed + deterministic execution:

- classification (ResNet-50, L-WISE eps1): `max|Δ| = 0`, SHA256 match
- detection (Faster R-CNN, on the `ScalarDetectionWrapper` scalar): `max|Δ| = 0`, SHA256 match

So this issue is about **locking the property into CI**, not fixing a regression — and it closes the one gap in the parity suite.
